metashare/api/validators.py

```diff
@@ -49,9 +49,13 @@ def filter_queryset(self, attrs, queryset, serializer):
 )
 
 
-def repository_github_user_validator(cleaned_data):
-    repository = cleaned_data["repository"]
-    github_users = cleaned_data["github_users"]
-    for user in github_users:
-        if user not in repository.github_users:
-            raise ValidationError(f"Invalid github_users value: {user}")
+class GitHubUserValidator:
+    def __init__(self, *, parent):
+        self.parent = parent
+
+    def __call__(self, cleaned_data):
+        parent = cleaned_data.get(self.parent)
+        github_users = cleaned_data.get("github_users", [])
+        for user in github_users:
+            if user not in parent.github_users:
+                raise ValidationError(f"Invalid github_users value: {user}")
```

Comment:
@wlonk This validator is still (again) only used for projects, so we could revert this change if you like.

Reply:
I'm fine with this as-is. The generalization was motivated by code, even if the code is pulled back from there now, and so we understand the problem space enough to justify it.